new API pdfToHtmlRetrieve to convert PDFs in the Chrome Extension to HTML files in S3

SailingSteve opened the pull request:
Reads in the PDF by URL, applies it to pdf2Text.py by calling the Python tool https://github.com/pdfminer/pdfminer.six  (There are many other forks of pdfminer in Github, but the one we use is pdfminer.six, not plain pdfminer!)
Files in s3 are set to expire in 1 week
A good test URL:
https://www.iuoe399.org/media/filer_public/45/77/457700c9-dd70-4cfc-be49-a81cb3fba0a6/2020_lu399_primary_endorsement.pdf
Becomes:
https://wevote-temporary.s3.amazonaws.com/tmprts8w5ub.html

SailingSteve committed:
new API pdfToHtmlRetrieve
Reads in the PDF by URL, applies it to pdf2Text.py by exec'ing a linux subprocess, using the Python tool https://github.com/pdfminer/pdfminer.six  (There are many other forks, but the one we use is pdfminer.six, not plain pdfminer!)
A good test URL:
https://www.iuoe399.org/media/filer_public/45/77/457700c9-dd70-4cfc-be49-a81cb3fba0a6/2020_lu399_primary_endorsement.pdf

SailingSteve committed:
new API pdfToHtmlRetrieve
Reads in the PDF by URL, applies it to pdf2Text.py by exec'ing a linux subprocess, using the Python tool https://github.com/pdfminer/pdfminer.six  (There are many other forks, but the one we use is pdfminer.six, not plain pdfminer!)
A good test URL:
https://www.iuoe399.org/media/filer_public/45/77/457700c9-dd70-4cfc-be49-a81cb3fba0a6/2020_lu399_primary_endorsement.pdf

DaleMcGrew commented:
Thank you @SailingSteve!  I'm wondering if we can add "pdfminer.six" to our WeVoteServer/requirements.txt file so that our production servers just pick it up instead of having to add it manually?

@josephevans do you have any thoughts about automating Steve's instructions (in this pull request) re: creating a symlink on the live servers? 
```Then in a terminal window, create a symlink to the executable script pdf2txt.py```

SailingSteve committed:
Added a new API pdfToHtmlRetrieve, allows the chrome extension to call an API to convert PDFs to HTML
Reads in the PDF by URL, applies it to pdf2Text.py by calling the Python tool https://github.com/pdfminer/pdfminer.six  (There are many other forks, but the one we use is pdfminer.six, not plain pdfminer!)
A good test URL:
https://www.iuoe399.org/media/filer_public/45/77/457700c9-dd70-4cfc-be49-a81cb3fba0a6/2020_lu399_primary_endorsement.pdf

DaleMcGrew commented:
👍
